src/client/datascience/kernel-launcher/kernelLauncher.ts

```diff
@@ -166,7 +176,7 @@ export class KernelLauncher implements IKernelLauncher {
         const port = nextFreePort > startPort + 1_000 ? startPort : nextFreePort;
 
         // Then get the next set starting at that point
-        const ports = await getPorts(5, { host: '127.0.0.1', port });
+        const ports = await KernelLauncher.findNextFreePort(port);
 
         // We launch restart kernels in the background, its possible other session hasn't started.
         // Ensure we do not use same ports.
```

Comment:
You shouldn't need nextFreePortToTryAndUsePromise anymore

Reply:
Reverted the change, & will fix it separately